components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/utils/AuthUtil.java

```diff
@@ -56,7 +56,7 @@ public static String getHttpOnlyCookieHeader(Cookie cookie) {
 
     public static String getAppContext(Request request) {
         //TODO this method should provide uuf app context. Consider the scenarios of reverse proxy as well.
-        return "/" + request.getProperty("REQUEST_URL").toString().split("/")[1];
+        return "/" + request.getProperty("REQUEST_URL").toString().split("/")[3];
```

Comment:
I don't think this code will work when you reverse proxy the application. If you have a custom URL this logic will fail.